client/core/types.go

```diff
@@ -685,6 +699,11 @@ func (a *dexAccount) setupCryptoV2(creds *db.PrimaryCredentials, crypter encrypt
 
 func (a *dexAccount) resetKeys() {
 	a.keyMtx.Lock()
+	if len(a.encKey) > 0 {
+		// TODO: Should only increment if the current keys were derived
+		// from a seed.
```

Comment:
This would be unnecessary if we decide to only generate hd accounts going forward.